zrml/orderbook/src/types.rs

```diff
@@ -29,7 +29,7 @@ pub enum OrderSide {
 }
 
 #[derive(Clone, Encode, Eq, Decode, MaxEncodedLen, PartialEq, RuntimeDebug, TypeInfo)]
-pub struct Order<AccountId, Balance, MarketId: MaxEncodedLen> {
+pub struct Order<AccountId, Balance, MarketId: MaxEncodedLen + HasCompact> {
```

Comment:
Storage migration?

Reply:
Not necessary, only new yet unused types utilize the compact market id.